From 49e9b2a1dd9b7c7fd278363da3947b90986ca718 Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Sun, 18 Jun 2017 00:37:39 +0800
Subject: [PATCH 01/47] gallium/u_math: add ushort_to_float/float_to_ushort

Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 src/gallium/auxiliary/util/u_math.h | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/src/gallium/auxiliary/util/u_math.h b/src/gallium/auxiliary/util/u_math.h
index 79869a1..e80c6d7 100644
--- a/src/gallium/auxiliary/util/u_math.h
+++ b/src/gallium/auxiliary/util/u_math.h
@@ -390,6 +390,37 @@ float_to_ubyte(float f)
    }
 }
 
+/**
+ * Convert ushort to float in [0, 1].
+ */
+static inline float
+ushort_to_float(ushort us)
+{
+   return (float) us * (1.0f / 65535.0f);
+}
+
+
+/**
+ * Convert float in [0,1] to ushort in [0,65535] with clamping.
+ */
+static inline ushort
+float_to_ushort(float f)
+{
+   union fi tmp;
+
+   tmp.f = f;
+   if (tmp.i < 0) {
+      return (ushort) 0;
+   }
+   else if (tmp.i >= 0x3f800000 /* 1.0f */) {
+      return (ushort) 65535;
+   }
+   else {
+      tmp.f = tmp.f * (65535.0f/65536.0f) + 128.0f;
+      return (ushort) tmp.i;
+   }
+}
+
 static inline float
 byte_to_float_tex(int8_t b)
 {
-- 
2.0.1

