diff --git a/Makefile b/Makefile
index 4cb6cde..565962d 100644
--- a/Makefile
+++ b/Makefile
@@ -587,7 +587,6 @@ include $(srctree)/scripts/Makefile.extrawarn
 KBUILD_CPPFLAGS += $(KCPPFLAGS)
 KBUILD_AFLAGS += $(KAFLAGS)
 KBUILD_CFLAGS += $(KCFLAGS)
-KBUILD_CFLAGS += -Werror
 
 # Use UBOOTINCLUDE when you must reference the include/ directory.
 # Needed to be compatible with the O= option
@@ -863,14 +862,6 @@ FIP_ARGS += --bl33 $(FIP_FOLDER_SOC)/bl33.bin
 
 .PHONY: fip_create
 fip_create:
-	$(Q)$(FIP_FOLDER)/fip_create > /dev/null 2>&1 && \
-		echo "Pre-built fip_create is valid" || \
-		echo "Re-build fip_create"
-ifneq ($$?, "0")
-	$(Q)rm -f $(FIP_FOLDER)/fip_create
-	$(Q)$(MAKE) -C $(srctree)/tools/fip_create/
-	$(Q)cp $(srctree)/tools/fip_create/fip_create $(FIP_FOLDER)/
-endif
 
 .PHONY: fip.bin
 fip.bin: tools prepare u-boot.bin bl301.bin fip_create
@@ -1416,7 +1407,6 @@ distclean: mrproper
 	@rm -f $(FIP_FOLDER_SOC)/u-boot.bin $(srctree)/u-boot.bin
 	@rm -f $(FIP_FOLDER_SOC)/u-boot.bin.* $(FIP_FOLDER_SOC)/*.encrypt
 	@rm -f $(FIP_FOLDER)/u-boot.bin.* $(FIP_FOLDER)/*.bin $(FIP_FOLDER)/*.encrypt
-	@rm -f $(FIP_FOLDER)/fip_create
 	@rm -f $(srctree)/fip/aml_encrypt_gxb
 	@rm -f $(FUSING_FOLDER)/u-boot.bin
 	@$(MAKE) -C $(srctree)/tools/fip_create clean
