diff --git a/Makefile b/Makefile
index 28abd03..8408e76 100644
--- a/Makefile
+++ b/Makefile
@@ -564,6 +564,8 @@ ifeq ($(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-stack-usage.sh $(CC)),y)
 endif
 
 KBUILD_CFLAGS += $(call cc-option,-Wno-format-nonliteral)
+KBUILD_CFLAGS += $(call cc-option,-Wno-misleading-indentation)
+KBUILD_CFLAGS += $(call cc-option,-Wno-tautological-compare)
 
 # turn jbsr into jsr for m68k
 ifeq ($(ARCH),m68k)
