From 2f71523704c2eb75b461b36f806932299ba1bf2a Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Tue, 9 May 2017 21:26:29 +0800
Subject: [PATCH] arm/dts/sun8i-h3: add GPU mali400 node

Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 arch/arm/boot/dts/sun8i-h3.dtsi | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-h3.dtsi b/arch/arm/boot/dts/sun8i-h3.dtsi
index 8495dee..3c690b6 100644
--- a/arch/arm/boot/dts/sun8i-h3.dtsi
+++ b/arch/arm/boot/dts/sun8i-h3.dtsi
@@ -79,6 +79,36 @@
 			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>;
 	};
+
+	soc {
+		gpu@01c40000 {
+			compatible = "arm,mali400";
+			reg = <0x01c40000 0x1000
+			       0x01c41000 0x1000
+			       0x01c42000 0x1000
+			       0x01c43000 0x1000
+			       0x01c44000 0x1000
+			       0x01c45000 0x1000
+			       0x01c48000 0x2000
+			       0x01c4a000 0x2000>;
+			reg-names = "gp", "l2-cache", "pmu", "gp-mmu", "pp0-mmu",
+				    "pp1-mmu", "pp0", "pp1";
+			interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
+					   <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "gp", "gp-mmu", "pp0", "pp0-mmu", "pmu",
+					  "pp1", "pp1-mmu";
+			clocks = <&ccu CLK_BUS_GPU>, <&ccu CLK_GPU>;
+			clock-names = "bus", "gpu";
+			resets = <&ccu RST_BUS_GPU>;
+			reset-names = "ahb";
+			num-pp = <2>;
+		};
+	};
 };
 
 &ccu {
