From e41109a99bd5a4f97ea7834f999f268958ca6593 Mon Sep 17 00:00:00 2001
From: Neil Armstrong <narmstrong@baylibre.com>
Date: Thu, 18 Jan 2018 15:45:52 +0100
Subject: [PATCH] drm: lima: also allow on ARCH_MESON

---
 drivers/gpu/drm/lima/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/lima/Kconfig b/drivers/gpu/drm/lima/Kconfig
index 03501ce..e963e96 100644
--- a/drivers/gpu/drm/lima/Kconfig
+++ b/drivers/gpu/drm/lima/Kconfig
@@ -2,7 +2,7 @@
 config DRM_LIMA
        tristate "LIMA (DRM support for ARM Mali 400 GPU)"
        depends on DRM
-       depends on ARCH_SUNXI || ARCH_ROCKCHIP || ARCH_EXYNOS
+       depends on ARCH_SUNXI || ARCH_ROCKCHIP || ARCH_EXYNOS || ARCH_MESON
        select INTERVAL_TREE
        help
          DRM driver for ARM Mali 400 GPUs.
