From 2c448207bc24c59dd118c0f495d9675fe0cc30e7 Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Fri, 2 Feb 2018 09:20:25 +0800
Subject: [PATCH] drm/lima: change submit ABI for better compatibility

Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 include/uapi/drm/lima_drm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/drm/lima_drm.h b/include/uapi/drm/lima_drm.h
index 45fcd26..bd0cc05 100644
--- a/include/uapi/drm/lima_drm.h
+++ b/include/uapi/drm/lima_drm.h
@@ -134,9 +134,9 @@ struct drm_lima_gem_submit {
 	__u32 fence;       /* out */
 	__u32 pipe;        /* in */
 	__u32 nr_bos;      /* in */
+	__u32 frame_size;  /* in */
 	__u64 bos;         /* in */
 	__u64 frame;       /* in */
-	__u64 frame_size;  /* in */
 };
 
 struct drm_lima_wait_fence {
